rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RealPreference.java

```diff
@@ -65,22 +67,21 @@
     return adapter.get(key, preferences);
   }
 
-  @Override public synchronized void set(@Nullable T value) {
+  @Override public synchronized void set(@NonNull T value) {
+    checkNotNull(value, "value == null");
     SharedPreferences.Editor editor = preferences.edit();
-    if (value == null) {
-      editor.remove(key);
-    } else {
-      adapter.set(key, value, editor);
-    }
+    adapter.set(key, value, editor);
     editor.apply();
   }
 
-  @Override public boolean isSet() {
+  @Override public synchronized boolean isSet() {
```

Comment:
Okay then I get it. The synchronized is to not confront the adapter with a situation where the value is absent. So I left it only for `delete` and `get`